### PR TITLE
Table of Contents for README

### PR DESCRIPTION
**Pull request by vandyliu**
[AB#34667](https://microsoftgarage.visualstudio.com/002806e2-ebaa-4672-9d2e-5fe5d29154ef/_workitems/edit/34667)

# Description:
- Added Table of Contents for README
- Purposely didn't include some parts because I didn't want to clutter the table of contents (eg. build status, every step for how to use) but let me know your opinions.

# Limitations:

Please describe limitations of this PR

# Testing:

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] My code has been formatted with `npm run format` and passes the checks in `npm run check`
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by nasadigital — APPROVED** [review]
This is great!
LGTM

File: README.md [hunk 1]
```diff
@@ -8,7 +8,32 @@ Make without limit! Device Simulator Express, a Microsoft Garage project, allows
 result when you plug in your actual microcontroller. Curious about the output of the device, the serial
 monitor allows you to observe the device output.
 
-## Devices we support:
+## Table of Contents
+  - [Devices we support](#devices-we-support)
```
Comment (nasadigital) [inline README.md:12]: Do we know how the table of contents will work in the marketplace?

Comment (vandyliu) [reply README.md:12]: It works! The links are markdown links and we have some in the marketplace already for "devices we support". Clicking on a device takes you to the device part in the marketplace :)

**Review by xnkevinnguyen — APPROVED** [review]
THANK YOU! much needed

**Pull request merged**